include/mfmg/utils.cuh

```diff
@@ -0,0 +1,146 @@
+/*************************************************************************
+ * Copyright (c) 2017 by the mfmg authors                                *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef UTILS_CUH
+#define UTILS_CUH
+
+#ifdef MFMG_WITH_CUDA
+
+#include <mfmg/exceptions.hpp>
+
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/trilinos_index_access.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+
+#include <tuple>
+
+namespace mfmg
+{
+namespace internal
+{
+template <typename ScalarType>
+std::tuple<ScalarType *, int *, int *>
+copy_to_gpu(std::vector<ScalarType> const &val,
```

Comment:
Maybe replace these function by a simpler one?
```
template<typename Type>
Type* copy_to_gpu(std::vector<Type> const& vec) {
}
```
These would reduce copy-paste code.

Reply:
I am not sure what you mean. There is only one function `copy_to_gpu`.

Reply:
In your `copy_to_gpu` you provide three arrays of different types, and return a tuple of pointers. I propose modifying it that instead it takes in a single array, and being called three times.